helion/_compiler/inductor_lowering.py

```diff
@@ -1048,13 +1048,14 @@ def apply_dot_requirements(
 def reduce_3d_dot(
     ctx: GraphInterpreter, node: torch.fx.Node, with_acc: bool
 ) -> ast.AST:
-    datatype = CompileEnvironment.current().settings.dot_precision
     acc = None
+    acc_node: torch.fx.Node | None = None
     if with_acc:
         acc, lhs, rhs = map_arg(node.args, lambda arg: ctx.env[arg])
         assert isinstance(acc, ast.AST)
         lhs_node = node.args[1]
         rhs_node = node.args[2]
+        acc_node = node.args[0] if isinstance(node.args[0], torch.fx.Node) else None
```

Comment:
Should the `if isinstance(node.args[0], torch.fx.Node)` be an assert?

I worry this will hide a tricky bug where `node.args[0]` is something else other than Node/None.